File: LeetCode/cpp/1347_Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram.cpp
class Solution {
public:
	int minSteps(string s, string t) {
		int answer = 0;

		unordered_map<char, int> sMap, tMap;

		for (int i = 0; i < s.length(); i++) {
			sMap[s[i]]++;
			tMap[t[i]]++;
		}

		for (auto map : tMap) {
			if (sMap.count(map.first) == 1) {
				if (sMap[map.first] <= tMap[map.first]) {
					sMap.erase(map.first);
				}
				else {
					sMap[map.first] = sMap[map.first] - tMap[map.first];
				}
			}
		}

		for (auto map : sMap) {
			answer += sMap[map.first];
		}

		return answer;
	}
};

File: LeetCode/java/1347_Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int minSteps(String s, String t) {
        int answer = 0;

        Map<Character, Integer> sMap = new HashMap<>();
        Map<Character, Integer> tMap = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            sMap.put(s.charAt(i), sMap.getOrDefault(s.charAt(i), 0) + 1);
            tMap.put(t.charAt(i), tMap.getOrDefault(t.charAt(i), 0) + 1);
        }

        for (char key : tMap.keySet()) {
            if (sMap.containsKey(key)) {
                if (sMap.get(key) <= tMap.get(key)) {
                    sMap.remove(key);
                } else {
                    sMap.put(key, sMap.get(key) - tMap.get(key));
                }
            }
        }

        for (char key : sMap.keySet()) {
            answer += sMap.get(key);
        }

        return answer;
    }
}